azurerm/helpers/azure/container_group.go

```diff
@@ -0,0 +1,91 @@
+package azure
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/containerinstance/mgmt/2018-10-01/containerinstance"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+)
+
+func SchemaContainerGroupProbe() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		ForceNew: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"exec": {
+					Type:     schema.TypeList,
+					Optional: true,
+					ForceNew: true,
+					Elem: &schema.Schema{
+						Type:         schema.TypeString,
+						ValidateFunc: validation.NoZeroValues,
+					},
+				},
+
+				"httpget": {
+					Type:     schema.TypeList,
+					Optional: true,
+					ForceNew: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"path": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ForceNew:     true,
+								ValidateFunc: validate.NoEmptyStrings,
+							},
+							"port": {
+								Type:         schema.TypeInt,
+								Optional:     true,
+								ForceNew:     true,
+								ValidateFunc: validate.PortNumber,
+							},
+							"scheme": {
+								Type:     schema.TypeString,
+								Optional: true,
+								ForceNew: true,
+								ValidateFunc: validation.StringInSlice([]string{
+									string(containerinstance.HTTP),
+									string(containerinstance.HTTPS),
+								}, true),
```

Comment:
Could we change this ti false? we are moving towards case sensitivity for properties like this

Reply:
@katbyte I am struggling a bit with this one. The Azure API always returns `Http` or `Https`. When setting `ignoreCase` to `false` tests on this value begin to fail. 

My assumption is that these tests should work.

```
    liveness_probe {
      httpget {
        path   = "/"
        port   = 443
        scheme = "Http"
      }
      inital_delay_seconds = 1
      period_seconds       = 1
      failure_threashold   = 1
      sucess_threashold    = 1
      timeout_seconds      = 1
    }
```

and then:

```
resource.TestCheckResourceAttr(resourceName,"container.0.liveness_probe.0.httpget.0.scheme", "Http"),
```

However, it seems I am hitting [this](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/vendor/github.com/hashicorp/terraform/helper/validation/validation.go#L90), which returns:

```
testing.go:538: Step 0 error: config is invalid: azurerm_container_group.test: expected container.0.liveness_probe.0.httpget.0.scheme to be one of [http https], got Http
```

I am still investigating but if you have any advice, I would appreciate it :).

Thanks

Reply:
Thanks, Tom. I figured something was going on here and had considered this approach. Appreciate the assist and validation.